R/03-modelEstimation.R

```diff
@@ -339,3 +384,51 @@ modelEstimation <- function(input, output, session, data) {
     }
   })
 }
+
+keep_selected <- function(selected, choices) {
+  if (is.null(choices) || length(choices) == 0) return(character(0))
+  if (is.null(selected)) return(character(0))
+  
+  sel <- selected
+  sel <- sel[sel %in% choices]                     # keep only valid choices
+  sel <- if (length(sel)) sel else character(0)    # fall back if nothing left
+  sel
+}
+
+# Returns the INVALID terms from `ret` based on:
+# - any inverse power I(var^k) with k < 0 AND the column has zeros
+# - any exponent I(var^k) where the column is non-numeric
+# - any exponent I(var^k) where var is missing or exponent can't be parsed
```

Comment:
@mgross now we check if a term is valid. Do you see any other cases that we should consider here, except of 
- zeros in columns with a negative power, or
- non-numeric columns with powers?